test/e2e/node_feature_discovery_test.go

```diff
@@ -909,6 +912,73 @@ denyLabelNs: []
 					)).NotTo(HaveOccurred())
 				})
 			})
+
+			Context("and test whether resyncPeriod is passed successfully or not", func() {
+				BeforeEach(func(ctx context.Context) {
+					extraMasterPodSpecOpts = []testpod.SpecOption{
+						testpod.SpecWithConfigMap("nfd-master-conf", "/etc/kubernetes/node-feature-discovery"),
+					}
+					cm := testutils.NewConfigMap("nfd-master-conf", "nfd-master.conf", `
+resyncPeriod: "1s"
+`)
+					_, err := f.ClientSet.CoreV1().ConfigMaps(f.Namespace.Name).Create(ctx, cm, metav1.CreateOptions{})
+					Expect(err).NotTo(HaveOccurred())
+				})
+				It("labels should be restored to the original ones", func(ctx context.Context) {
+					// deploy node feature object
+					if !useNodeFeatureApi {
+						Skip("NodeFeature API not enabled")
+					}
+
+					nodes, err := getNonControlPlaneNodes(ctx, f.ClientSet)
+					Expect(err).NotTo(HaveOccurred())
+
+					targetNodeName := nodes[0].Name
+					Expect(targetNodeName).ToNot(BeEmpty(), "No suitable worker node found")
+
+					// Apply Node Feature object
+					By("Creating NodeFeature object")
+					nodeFeatures, err := testutils.CreateOrUpdateNodeFeaturesFromFile(ctx, nfdClient, "nodefeature-1.yaml", f.Namespace.Name, targetNodeName)
+					Expect(err).NotTo(HaveOccurred())
+
+					By("Verifying node labels from NodeFeature object #1")
+					expectedLabels := map[string]k8sLabels{
+						targetNodeName: {
+							nfdv1alpha1.FeatureLabelNs + "/e2e-nodefeature-test-1": "obj-1",
+							nfdv1alpha1.FeatureLabelNs + "/e2e-nodefeature-test-2": "obj-1",
+							nfdv1alpha1.FeatureLabelNs + "/fake-fakefeature3":      "overridden",
+						},
+					}
+					Expect(checkForNodeLabels(ctx, f.ClientSet,
+						expectedLabels, nodes,
+					)).NotTo(HaveOccurred())
+
+					patches, err := json.Marshal(
+						[]apihelper.JsonPatch{
+							apihelper.NewJsonPatch(
+								"replace",
+								"/metadata/labels",
+								nfdv1alpha1.FeatureLabelNs+"/e2e-nodefeature-test-1",
+								"randomValue",
+							),
+						},
+					)
+					Expect(err).NotTo(HaveOccurred())
+
+					_, err = f.ClientSet.CoreV1().Nodes().Patch(context.TODO(), targetNodeName, types.JSONPatchType, patches, metav1.PatchOptions{})
```

Comment:
Let's use proper context as we have one
```suggestion
					_, err = f.ClientSet.CoreV1().Nodes().Patch(ctx, targetNodeName, types.JSONPatchType, patches, metav1.PatchOptions{})
```